src/arrayMethodJoin.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let actualSep = separator;
+
+    if (separator === undefined) {
+      actualSep = ',';
+    }
```

Comment:
Instead of doing this, you could just write separator = ',' inside function parentheses.